lab_5_scrapper/scrapper.py

```diff
@@ -90,25 +204,41 @@ def __init__(self, config: Config) -> None:
         """
         Initializes an instance of the Crawler class
         """
-        pass
+        self.urls = []
+        self.config = config
+        self.seed_urls = config.get_seed_urls()
+        self.number_of_articles = config.get_num_articles()
 
     def _extract_url(self, article_bs: BeautifulSoup) -> str:
         """
         Finds and retrieves URL from HTML
         """
-        pass
+        all_links_bs = article_bs.find_all('a')
+        for link in all_links_bs:
+            href = link.get('href')
```

Comment:
who guarantees that there is always string? otherwise next line is going to fail in case of None